test/integration/grpc/grpc_config_apply_test.go

```diff
@@ -0,0 +1,159 @@
+package grpc
+
+import (
+	"net"
+	"os"
+	"testing"
+
+	"google.golang.org/grpc"
+
+	sdkGRPC "github.com/nginx/agent/sdk/v2/grpc"
+	sdkProto "github.com/nginx/agent/sdk/v2/proto"
+	"github.com/nginx/agent/test/integration/utils"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+
+	agent_config "github.com/nginx/agent/sdk/v2/agent/config"
+)
+
+const (
+	GRPC_ADDRESS  = ":54789"
```

Comment:
Optional:

I think if we hit the `panic()` for binding to this port, the deferred close of the listener will never get executed and will likely panic on subsequent runs.

One potential way to reduce this minor risk further would be be us the special value `:0` which will bind to a random available port. (Assuming port number is arbitrary)

Ref: https://pkg.go.dev/net#Listen

Reply:
port number is used in the agent config in the container so it needed to be known before tests are run.
https://github.com/nginx/agent/blob/672bcf6183f24ed10db281105aef5ffd3c12eac2/test/integration/grpc/nginx-agent.conf#L12

Reply:
Made the port dynamic now.